http-client/src/main/java/com/facebook/airlift/http/client/jetty/ConcurrentScheduler.java

```diff
@@ -1,61 +1,57 @@
 package com.facebook.airlift.http.client.jetty;
 
+import com.facebook.airlift.concurrent.ConcurrentScheduledExecutor;
 import org.eclipse.jetty.util.component.AbstractLifeCycle;
 import org.eclipse.jetty.util.thread.Scheduler;
 
+import java.util.Optional;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.ScheduledFuture;
-import java.util.concurrent.ScheduledThreadPoolExecutor;
-import java.util.concurrent.ThreadFactory;
-import java.util.concurrent.ThreadLocalRandom;
 import java.util.concurrent.TimeUnit;
 
-import static com.facebook.airlift.concurrent.Threads.daemonThreadsNamed;
 import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
-// based on ScheduledExecutorScheduler
-class ConcurrentScheduler
+public class ConcurrentScheduler
         extends AbstractLifeCycle
         implements Scheduler
 {
+    private Optional<ConcurrentScheduledExecutor> concurrentScheduler;
```

Comment:
Does this optional make sense when we don't check if it's empty before using it?

Reply:
It is checked in doStop()